lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { FocusEvent, KeyboardEvent, Ref, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import { always, dec, inc } from 'ramda';
+
+import './NxCombobox.scss';
+
+import forwardRef from '../../util/genericForwardRef';
+import { Props, propTypes } from './types';
+import NxTextInput from '../NxTextInput/NxTextInput';
+import NxDropdownMenu from '../NxDropdownMenu/NxDropdownMenu';
+import NxLoadError from '../NxLoadError/NxLoadError';
+import NxLoadingSpinner from '../NxLoadingSpinner/NxLoadingSpinner';
+import { useUniqueId } from '../../util/idUtil';
+import DataItem from '../../util/DataItem';
+export { Props } from './types';
+
+function NxComboboxRender<T extends string | number = string>(
+  props: Props<T>,
+  ref: Ref<HTMLDivElement>
+) {
+  const {
+        className: classNameProp,
+        loading,
+        loadError,
+        matches,
+        value,
+        onChange,
+        onSearch,
+        disabled,
+        emptyMessage,
+        autoComplete,
+        validatable,
+        isPristine,
+        validationErrors,
+        id,
+        'aria-required': ariaRequired,
+        'aria-describedby': ariaDescribedBy,
+        'aria-label': ariaLabel,
+        ...attrs
+      } = props,
+
+      isEmpty = !matches.length,
+      isAlert = loading || loadError || isEmpty,
+      dropdownRef = useRef<HTMLDivElement>(null),
+      inputRef = useRef<HTMLDivElement>(null),
+
+      [focusableBtnIndex, setFocusableBtnIndex] = useState<number | null>(null),
+      [showDropdown, setShowDropdown] = useState(false),
+      [inlineStyle, setInlineStyle] = useState(false),
+      inputVal = autoComplete && focusableBtnIndex !== null ? matches[focusableBtnIndex].displayName : value,
+
+      inputId = useUniqueId('nx-combobox-input', id),
+      alertDropdownId = useUniqueId('nx-combobox-alert-dropdown'),
+      dropdownId = useUniqueId('nx-combobox-dropdown'),
+      dropdownBtnId = useUniqueId('nx-dropdown-button'),
```

Comment:
Can we call this `dropdownBtnIdPrefix` since it isn't used as an actual id by itself?

Reply:
Updated here 39113a22b610af972d3f1832523656328201e6ee.